src/win/getaddrinfo.c

```diff
@@ -126,34 +115,44 @@ static void uv__getaddrinfo_done(struct uv__work* w, int status) {
   }
 
   if (req->retcode == 0) {
+    char *alloc_ptr = NULL;
+    size_t cur_off = 0;
+    size_t addrinfo_len;
     /* Convert addrinfoW to addrinfo. First calculate required length. */
-    addrinfow_ptr = req->addrinfow;
+    struct addrinfoW *addrinfow_ptr = req->addrinfow;
     while (addrinfow_ptr != NULL) {
-      addrinfo_len += addrinfo_struct_len +
-          ALIGNED_SIZE(addrinfow_ptr->ai_addrlen);
+      align_offset(&cur_off, sizeof(void*));
+      cur_off += sizeof(struct addrinfo);
+      /* TODO: This alignment could be smaller, if we could
+	           portably get the alignment for sockaddr. */
+      align_offset(&cur_off, sizeof(void*));
+      cur_off += addrinfow_ptr->ai_addrlen;
```

Comment:
Arguably easier to read if you make `align_offset()` return the new value instead of mutating it in place:
```suggestion
      cur_off += align_offset(cur_off, sizeof(void*));
      cur_off += sizeof(struct addrinfo);
      /* TODO: This alignment could be smaller, if we could
	           portably get the alignment for sockaddr. */
      cur_off += align_offset(cur_off, sizeof(void*));
      cur_off += addrinfow_ptr->ai_addrlen;
```
I don't understand the TODO, btw. I'm also not 100% sure the alignment computations here are correct but I don't know exactly how the objects are laid out in memory.

@libuv/windows you should take a careful look at this

Reply:
Here we are pessimizing by assuming sockaddr needs pointer alignment, but I have seen that on some platforms it needs only uint16_t alignment or so.

The problem is that C89 does not have a ready made facility for obtaining alignment requirement of a given type.
I felt introducing that on this patch would be bigger change with more involved review than needed, considering existing code already suffers from this limitation, and if we introduced such a facility, there likely would be many other unrelated areas of libuv we would want to use it on.

Reply:
I would guess that C89 can express it as an offsetof computation, roughly as follows: `alignof(t)  = offsetof(struct { char x; t y }, y)` but it does seem pretty annoying

Reply:
That does not work when compiling as C++, and as C it does, but GCC gives bogus warning about missing semicolon at the end of the struct definition. But you can't possibly put a semicolon there anyway: https://godbolt.org/z/GYWorbKMY

Reply:
A bunch of other C compilers don't accept this as well. MSVC, Compcert, tiny c, sdcc, to name some.
https://godbolt.org/z/GYWorbKMY